static/panes/editor.js

```diff
@@ -500,6 +519,55 @@ Editor.prototype.updateOpenInQuickBench = function () {
         text: this.getSource()
     };
 
+    var compilers = this.getCompilerStates();
+
+    _.each(compilers, function (compiler) {
+        var knownCompiler = false;
+        if (compiler.compiler.match(/^g[0-9]*$/)) {
+            knownCompiler = true;
+            quickBenchState.compiler = "gcc-" + compiler.compiler[1] + "." + compiler.compiler[2];
+        } else if (compiler.compiler.match(/^clang[0-9]*x?$/)) {
+            knownCompiler = true;
+            quickBenchState.compiler = "clang-" + compiler.compiler[5] + "." + compiler.compiler[6];
+        }
+
+        if (knownCompiler) {
+            if (compiler.options.indexOf("-O3") !== -1) {
```

Comment:
```javascript
var match = compiler.options.match(/-(O([0-3sg]|fast))/);
if (match !== null)
  quickBenchState.optim = match[1];
```

regex not tested, but something like that could collapse these if chains down to a single statement
same for the cppVersion stuff below, although you would have to have to handle the edge cases for `2a`